lib/active_record/connection_adapters/oracle_enhanced/schema_statements.rb

```diff
@@ -156,6 +156,18 @@ def indexes(table_name) #:nodoc:
           all_schema_indexes.select { |i| i.table == table_name }
         end
 
+        def columns(table_name)
+          table_name = table_name.to_s
+          if @columns_cache[table_name]
+            @columns_cache[table_name]
+          else
+            @columns_cache[table_name] =
+              column_definitions(table_name).map do |field|
+              new_column_from_field(table_name, field)
+            end
+          end
+          @columns_cache[table_name]
+        end
```

Comment:
How about using `super` to transfer to `AR::ConnectionAdapters::SchemaStatements#columns(table_name)` as follows?

```diff
         def columns(table_name)
           table_name = table_name.to_s
           if @columns_cache[table_name]
             @columns_cache[table_name]
           else
+            @columns_cache[table_name] = super(table_name)
-            @columns_cache[table_name] =
-              column_definitions(table_name).map do |field|
-              new_column_from_field(table_name, field)
-            end
           end
-          @columns_cache[table_name]
         end
```

There will be no duplication of implementation with super class.